### PR TITLE
feat(blocks): Add basic YouTube transcriber block

### PR DESCRIPTION
**Pull request by Torantulino**
Adds a basic YouTube Transcriber block. 

This implementation just uses the captions already present on the YouTube video.

In future this could be expanded to create new transcriptions, especially if they don't already exist on the video.

image

### CODE REVIEW

File: rnd/autogpt_server/pyproject.toml [hunk 1]
```diff
@@ -33,6 +33,7 @@ openai = "^1.35.7"
 jsonref = "^1.1.0"
 groq = "^0.8.0"
 anthropic = "^0.25.1"
+youtube-transcript-api = "^0.6.2"
```
Comment (majdyz) [inline rnd/autogpt_server/pyproject.toml:36]: This require poetry.lock update

File: rnd/autogpt_server/autogpt_server/blocks/youtube_transcriber.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            video_id = self.extract_video_id(input_data.youtube_url)
+            yield "video_id", video_id
+
+            transcript = self.get_transcript(video_id)
+            formatter = TextFormatter()
+            transcript_text = formatter.format_transcript(transcript)
+
+            yield "transcript", transcript_text
+        except Exception as e:
+            yield "error", str(e)
```
Comment (majdyz) [inline rnd/autogpt_server/autogpt_server/blocks/youtube_transcriber.py]: `poetry run format`

File: rnd/autogpt_server/autogpt_server/blocks/youtube_transcriber.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            output_schema=YouTubeTranscriber.Output,
+            test_input={
+                "youtube_url": "https://www.youtube.com/watch?v=dQw4w9WgXcQ"
+            },
+            test_output=[
+                ("video_id", "dQw4w9WgXcQ"),
+                ("transcript", "Never gonna give you up\nNever gonna let you down\n..."),
+            ],
+            test_mock={
+                "extract_video_id": lambda url: "dQw4w9WgXcQ",
```
Comment (majdyz) [inline rnd/autogpt_server/autogpt_server/blocks/youtube_transcriber.py]: Hmmm `extract_video_id` should not be mocked, no ?
We need to test the parsing is correct

File: rnd/autogpt_server/pyproject.toml (hunk 1 above)
Comment (Swiftyos) [reply rnd/autogpt_server/pyproject.toml:36]: Run `poetry lock` to do that

Comment (ntindle) [reply rnd/autogpt_server/pyproject.toml:36]: Done

File: rnd/autogpt_server/autogpt_server/blocks/youtube_transcriber.py (hunk 2 above)
Comment (ntindle) [reply rnd/autogpt_server/autogpt_server/blocks/youtube_transcriber.py]: Done

File: rnd/autogpt_server/autogpt_server/blocks/youtube_transcriber.py (hunk 3 above)
Comment (ntindle) [reply rnd/autogpt_server/autogpt_server/blocks/youtube_transcriber.py]: Done

**Review by aarushik93 — APPROVED** [review]

**Pull request merged**